optuna/distributions.py

```diff
@@ -280,36 +270,20 @@ class DiscreteUniformDistribution(BaseDistribution):
     """
 
     def __init__(self, low: float, high: float, q: float) -> None:
-        if low > high:
-            raise ValueError(
-                "The `low` value must be smaller than or equal to the `high` value "
-                "(low={}, high={}, q={}).".format(low, high, q)
-            )
-        if q <= 0:
-            raise ValueError("The `q` value must be non-zero positive value, but q={}.".format(q))
-
-        high = _adjust_discrete_uniform_high(low, high, q)
-
-        self.low = float(low)
-        self.high = float(high)
-        self.q = float(q)
-
-    def single(self) -> bool:
+        super().__init__(low=low, high=high, step=q)
+        self.q = q
```

Comment:
How about making this a property?
```python
@property
def q(self) -> int:
    return self.step
```

Reply:
It looks nice! I applied your suggestion in e4727cd. 👍